server/opts.go

```diff
@@ -1981,6 +1988,14 @@ func parseLeafNodes(v interface{}, opts *Options, errors *[]error, warnings *[]e
 		case "no_advertise":
 			opts.LeafNode.NoAdvertise = mv.(bool)
 			trackExplicitVal(opts, &opts.inConfig, "LeafNode.NoAdvertise", opts.LeafNode.NoAdvertise)
+		case "min_version", "minimum_version":
+			version := mv.(string)
+			if _, _, _, err := versionComponents(version); err != nil {
+				err := &configErr{tk, fmt.Sprintf("version parsing error: %v", err)}
+				*errors = append(*errors, err)
+				continue
+			}
```

Comment:
isn't this redundant with the code in validateLeafNode

Reply:
It is, and as you know, I often say that the check has to be done in "validate" so that it covers both config parsing and programmatic setting of the options. But when it is not too much to do, it is nice that it is done here too so that we print where in the config file the error is located, and also for the checking of the configuration file, command line `-t`, which otherwise says that the config is valid.
I am not saying that we should go back and do the checks in config parsing for all the ones that we did only in validate, but again, if "easy to do", then I think it is worth doing in both places. But I can remove if you think we should not have it in both places.

Reply:
ok with that